tests/Unit/DataStructures/Test_VectorImplHelper.hpp

```diff
@@ -177,5 +183,361 @@ auto remove_nth(const std::tuple<ValueTypes...>& tup) noexcept {
+  VectorType e1_copy_constructor{e1};  // NOLINT
+  CHECK(e1_copy_constructor.is_owning());
+  CHECK(e1_copy_constructor == e2);
+}
+
+/// \ingroup TestingFrameworkGroup
+/// \brief test the serialization of a `VectorType` constructed with a `ValType`
+template <typename VectorType, typename ValType>
+void vector_test_serialize(
+    typename tt::get_fundamental_type_t<ValType> low =
+    typename tt::get_fundamental_type_t<ValType>{-100},
+    typename tt::get_fundamental_type_t<ValType> high =
+    typename tt::get_fundamental_type_t<ValType>{100}) noexcept {
+  auto gen = make_generator();
+  UniformCustomDistribution<typename tt::get_fundamental_type_t<ValType>> dist{
+      low, high};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  const size_t npts = sdist(gen);
+  VectorType t(npts), tgood(npts);
+  VectorType t2;
+  auto start = make_with_random_values<ValType>(make_not_null(&gen),
+                                                make_not_null(&dist));
+  auto stride = make_with_random_values<ValType>(make_not_null(&gen),
+                                                 make_not_null(&dist));
+
+  ValType current = start;
+  std::generate(t.begin(), t.end(), [&current, stride ]() noexcept {
+    ValType ret = current;
+    current += stride;
+    return ret;
+  });
+  current = start;
+  std::generate(tgood.begin(), tgood.end(), [&current, stride ]() noexcept {
+    ValType ret = current;
+    current += stride;
+    return ret;
+  });
+  CHECK(tgood == t);
+  CHECK(t.is_owning());
+  CHECK(tgood.is_owning());
+  const VectorType serialized = serialize_and_deserialize(t);
+  CHECK(tgood == t);
+  CHECK(serialized == tgood);
+  CHECK(serialized.is_owning());
+  CHECK(serialized.data() != t.data());
+  CHECK(t.is_owning());
+  // checks serialization for reference
+  t2.set_data_ref(&t);
+  CHECK(t.is_owning());
+  CHECK_FALSE(t2.is_owning());
+  CHECK(t2 == t);
+  const VectorType serialized2 = serialize_and_deserialize(t2);
+  CHECK(t2 == t);
+  CHECK(serialized2 == t2);
+  CHECK(serialized2.is_owning());
+  CHECK(serialized2.data() != t2.data());
+  CHECK_FALSE(t2.is_owning());
+}
+
+/// \ingroup TestingFrameworkGroup
+/// \brief test the construction and movement of a reference `VectorType`
+/// constructed with a `ValType`
+template <typename VectorType, typename ValType>
+void vector_test_ref(
+    typename tt::get_fundamental_type_t<ValType> low =
+    typename tt::get_fundamental_type_t<ValType>{-100},
+    typename tt::get_fundamental_type_t<ValType> high =
+    typename tt::get_fundamental_type_t<ValType>{100}) noexcept {
+  auto gen = make_generator();
+  UniformCustomDistribution<typename tt::get_fundamental_type_t<ValType>> dist{
+      low, high};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t s1 = sdist(gen);
+  VectorType data(s1);
+  std::vector<typename VectorType::value_type> i(s1);
+  fill_with_random_values(make_not_null(&i), make_not_null(&gen),
```

Comment:
Put different construction techniques in their own tests, construct consistently elsewhere